hub/core/chunk.py

```diff
@@ -110,29 +193,65 @@ def register_sample_to_headers(
             ValueError: If `incoming_num_bytes` is not divisible by `num_samples`.
         """
 
-        num_bytes_per_sample = incoming_num_bytes
         self.shapes_encoder.register_samples(sample_shape, 1)
-        self.byte_positions_encoder.register_samples(num_bytes_per_sample, 1)
+        if (
+            incoming_num_bytes is not None
+        ):  # incoming_num_bytes is not applicable for non lz4 compressions
+            self.byte_positions_encoder.register_samples(incoming_num_bytes, 1)
+        self._clear_decompressed_cache()
 
     def update_sample(
-        self, local_sample_index: int, new_buffer: memoryview, new_shape: Tuple[int]
+        self,
+        local_sample_index: int,
+        new_buffer: memoryview,
+        new_shape: Tuple[int],
+        chunk_compression: Optional[str] = None,
```

Comment:
i don't know if we should have the chunk_compression be a parameter here do we? it is constant for a chunk/tensor

Reply:
It is but we dont want it to be a field in Chunk, because then we have to serialize it for caching to work.